vcs/gitcmd/repo.go

```diff
@@ -614,6 +628,20 @@ func (r *Repository) UpdateEverything(opt vcs.RemoteOpts) error {
 		cmd.Env = []string{"GIT_SSH=" + gitSSHWrapper}
 	}
 
+	if opt.HTTPS != nil {
```

Comment:
This would affect both opt.SSH and opt.HTTPS.

It could be, but it's not trivial because it would prevent `defer` from being used as it is used here now.

Additionally, this logic is really important and sensitive. I would prefer to avoid abstracting it away (making it harder to verify correctness), it should be more explicit what happens here. It's only repeated in two places (not 3 or more) at this time, so benefits from abstracting it are lesser.